src/radical/pilot/configs/resource_ornl.json

```diff
@@ -114,35 +126,43 @@
             "filesystem_endpoint"     : "file://localhost/"
         },
         "default_queue"               : "batch",
-        "resource_manager"            : "LSF_SUMMIT",
+        "resource_manager"            : "LSF",
+        "agent_config"                : "default",
         "agent_scheduler"             : "CONTINUOUS",
         "agent_spawner"               : "POPEN",
-        "agent_launch_method"         : "SSH",
-        "task_launch_method"          : "PRTE2",
-        "mpi_launch_method"           : "PRTE2",
-        "pre_bootstrap_0"             : ["module unload xl",
-                                         "module unload xalt",
-                                         "module unload spectrum-mpi",
-                                         "module load   gcc/6.4.0",
-                                         "module use    /sw/summit/ums/ompix/gcc/6.4.0/modules",
-                                         "module load   prrte/master",
-                                         "source /sw/summit/python/3.7/anaconda3/5.3.0/etc/profile.d/conda.sh"],
-        "pre_bootstrap_1"             : ["module unload xl",
-                                         "module unload xalt",
-                                         "module unload spectrum-mpi",
-                                         "module load   gcc/6.4.0",
-                                         "module use    /sw/summit/ums/ompix/gcc/6.4.0/modules",
-                                         "module load   prrte/master",
-                                         "ulimit -u 65536",
-                                         "source /sw/summit/python/3.7/anaconda3/5.3.0/etc/profile.d/conda.sh"],
+        "launch_methods"              : {
+                                         "order": ["PRTE", "SSH"],
+                                         # for sub-agent
+                                         "SSH"  : {},
+                                         "PRTE" : {
+                                             "pre_exec_cached": [
+                                                 "module unload xl",
+                                                 "module unload xalt",
+                                                 "module unload spectrum-mpi",
+                                                 "module load   gcc/6.4.0",
+                                                 "module use    /sw/summit/ums/ompix/gcc/6.4.0/modules",
+                                                 "module load   prrte/master"
+                                             ],
+                                             "pre_exec": [
+                                                 "ulimit -u 65536"
```

Comment:
should launch methods have `pre/post_launch` parameter? since `pre_exec` is what should be set within `exec_script` (which is executed with launch command)

As follow up to this I got confused (sorry if this was discussed already) with new `TaskDescription` attributes: `pre_launch` and `post_launch`. These attributes are related to LM and should be described in resource config (and `TaskDescription` is only responsible for task related attributes `pre/post_exec` and `pre/post_rank`)

Reply:
Second paragraph could be skipped (we discussed it - should be a possibility to set additional parameters if needed at every level)

First paragraph - we agreed to have `setup` or similar key instead of `pre_exec[_..]`, right?